slither/tools/read_storage/read_storage.py

```diff
@@ -240,16 +328,58 @@ def get_all_storage_variables(self, func: Callable = None) -> None:
             func (Callable, optional): A criteria to filter functions e.g. name.
         """
         for contract in self.contracts:
-            self._target_variables.extend(
-                filter(
-                    func,
-                    [
-                        (contract, var)
-                        for var in contract.state_variables_ordered
-                        if not var.is_constant and not var.is_immutable
-                    ],
-                )
-            )
+            for var in contract.state_variables_ordered:
+                if func is None or (func is not None and func(var)):
+                    if not var.is_constant and not var.is_immutable:
+                        self._target_variables.append((contract, var))
+                    elif var.is_constant and var.type == ElementaryType("bytes32"):
+                        self._constant_storage_slots.append((contract, var))
+            hardcoded_slot = self.find_hardcoded_slot_in_fallback(contract)
+            if hardcoded_slot is not None:
+                self._constant_storage_slots.append((contract, hardcoded_slot))
+
+    @staticmethod
+    def find_hardcoded_slot_in_fallback(contract: Contract) -> Optional[StateVariable]:
+        fallback = None
+        for func in contract.functions_entry_points:
+            if func.is_fallback:
+                fallback = func
+                break
+        if fallback is None:
+            return None
+        for node in fallback.all_nodes():
+            if node.type == NodeType.ASSEMBLY and isinstance(node.inline_asm, str):
+                asm_split = node.inline_asm.split("\n")
+                for asm in asm_split:
+                    if "sload(" in asm:
+                        arg = asm.split("sload(")[1].split(")")[0]
```

Comment:
I don't think this works if the argument is not a literal

Reply:
Yes, but finding that hardcoded Literal is the purpose of this function. I can't think of another circumstance where a constant slot isn't covered by either the for loop in `get_all_storage_variables` or the subsequent call to `find_hardcoded_slot_in_fallback`. If you think of one though I'd like to make sure it's covered!

Reply:
I mean that the parsing does not handle the case where it is something like `sload(add(1,1))`

Reply:
Good point. Granted, this part of the code (with the string parsing) is only meant for solc <0.6.0, where the assembly string is all we've got. Handling all of the corner cases with string parsing might get pretty ugly. However, the portion starting at line 364 with `elif node.type == NodeType.EPRESSION:` is meant to handle more recent solc versions, so this should definitely work if it's not a Literal. Let me test it a bit more.

Reply:
Given that this is already handling a bit of a corner case (usually storage slots are declared as constants), how robust does this need to be?

Reply:
The concern is that this breaks regular usage of slither-read-storage, so it should not fail if it comes across code it cannot parse

Reply:
Would it be sufficient for `find_hardcoded_slot_in_fallback` to return nothing if it finds code it can't parse?